lintro/ai/providers/cli_transport.py

```
@@ -0,0 +1,194 @@
+            if in_string:
+                continue
+            if char == "{":
+                depth += 1
+            elif char == "}":
+                depth -= 1
+                if depth == 0:
+                    return text[start : index + 1]
+        return text
+
+    @abstractmethod
+    def parse_stdout(self, stdout: str) -> Any:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `parse_stdout`

(ANN401)